File: python/Construct_the_Minimum_Bitwise_Array_II_3315.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        for i, num in enumerate(nums):
            if num % 2 == 0:
                nums[i] = -1
            else:
                nums[i] -= ((num + 1) & ~num) >> 1
        return nums

File: cpp/Construct_the_Minimum_Bitwise_Array_II_3315.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        for (int& num: nums) {
            if (num % 2 == 0) num = -1;
            else num -= (1 << __builtin_ctz(~num) - 1);
        }
        return nums;
    }
};
